### PR TITLE
Add filterable logging system to webapp using debug package

### PR DESCRIPTION
**Pull request by 5h1rU**
Closes https://github.com/sourcegraph/sourcegraph/issues/9944

If you want to enable the logger, please add `debug = *` in the browser's local storage.

Docs: https://www.npmjs.com/package/debug#web-browser

### CODE REVIEW

**Review by tjkandala — APPROVED** [review]
Looks good, here are some things we can add before merging:
- Recursive React hook to abstract over `Debugger#extend` for the web app
- Add at least root loggers for all processes (e.g. background page, worker thread, web app)

**Comment by marekweb** [comment]
Idea: it might be worthwhile to add a `sourcegraph:` prefix to the namespace. As we've seen from enabling the debug flag, various libraries use the `debug` library, so it would be helpful to be able to filter to the ones that comes from our own code.

**Comment by tjkandala** [comment]
Upon further investigation, we can't merge this PR, at least not with this library. We can't depend on `window.localStorage` in web workers, so we can't log extension actions now

**Review by tjkandala — CHANGES_REQUESTED** [review]
Upon further investigation, we can't merge this PR, at least not with this library. We can't depend on window.localStorage in web workers, so we can't log extension actions now

**Comment by felixfbecker** [comment]
There's a way to programmatically enable it, right? Can we use `postMessage()` from the main thread to enable it in the worker? Or, just use the comlink API setup we have in place? With the latter we could also easily make it work for the browser extension and background pages

File: yarn.lock [hunk 1]
```diff
@@ -19749,7 +19755,8 @@ sourcegraph@^24.0.0:
   integrity sha512-uB2SbjEzcA/HWP3Rj+INUXqQweO8HmRdVWW7zesFWZQFBcu7JTcq0pg0jeAN71glKPIyyOgZVrIm1d7KoVzfZw==
 
 "sourcegraph@link:client/packages/sourcegraph-extension-api":
-  version "24.8.0"
+  version "0.0.0"
```
Comment (umpox) [inline yarn.lock:19758]: 🤔

**Comment by Joelkw** [comment]
@muratsu how close to "done" is this now?

**Comment by muratsu** [comment]
We've shipped `sourcegraph.app.log` 4 days ago to aid debugging user issues. [#20474](https://github.com/sourcegraph/sourcegraph/pull/20474)

**Comment by 5h1rU** [comment]
Closing this by now. Let me know if you still find this valuable, and it will be open again.

**Comment by felixfbecker** [comment]
@5h1rU I still think this would have immense value for debugging purposes, especially debugging customer issues with CE. Loic mentioned they added something like this at his previous company and it was a game-changer.

**Pull request closed without merging**